### PR TITLE
Add Palette and VRAM

### LINKED ISSUE #86: Add Palette RAM
To manage correcty some mode of rendering of GBA Display we need to implement Palette RAM.
- [x] 0x05000000-0x050001FF - BG Palette RAM (512 bytes, 256 colors)   
- [x] 0x05000200-0x050003FF - OBJ Palette RAM (512 bytes, 256 colors)

### LINKED ISSUE #88: Add VRAM
First implementation of Video RAM. In future it can require some changes because data are structured in a certain way based on `MODE` of rendering.

### PR DESCRIPTION
**Pull request by gallottino**
Closes #86 
Closes #88

### CODE REVIEW

**Comment by gallottino** [comment]
I converted it to draft because I want to add also the issue #88

File: emu/src/internal_memory.rs [hunk 1]
```diff
@@ -217,4 +232,46 @@ mod tests {
...
+        im.write_at(address, 10);
+        assert_eq!(im.obj_palette_ram[8], 10);
+    }
+
+    #[test]
+    fn read_0bj_palette_ram() {
+        let mut im = InternalMemory::new();
+        im.obj_palette_ram[8] = 15;
+
+        let address = 0x05000208; // WININ lower byte
```
Comment (guerinoni) [inline emu/src/internal_memory.rs]: Is `// WININ lower byte` right comment here?

**Review by guerinoni — APPROVED** [review]
Looks good to me

File: emu/src/internal_memory.rs (hunk 1 above)
Comment (gallottino) [reply emu/src/internal_memory.rs]: No, just a misprint copy-paste. I'll remove it.

Comment (guerinoni) [reply emu/src/internal_memory.rs]: Oh nice :)

**Pull request merged**